holoviews/core/ndmapping.py

```diff
@@ -283,10 +279,9 @@ def groupby(self, dimensions, container_type=None, group_type=None, **kwargs):
         container_type = container_type if container_type else type(self)
         group_type = group_type if group_type else type(self)
         dimensions = [self.get_dimension(d, strict=True) for d in dimensions]
-        sort = not self._sorted
         with item_check(False):
             return util.ndmapping_groupby(self, dimensions, container_type,
-                                          group_type, sort=sort, **kwargs)
+                                          group_type, sort=True, **kwargs)
```

Comment:
Probably worth mentioning either in a comment - or maybe even the docstring - that sorting will always be applied.